### PR TITLE
Add CEReactions tests for HTMLButtonElement

### PR DESCRIPTION
**Pull request by xiaoyu3x**
- Verify attributes of HTMLButtonElement interface have CEReactions.
- Adjust functions ‘testReflectAttribute’ and ‘testReflectAttributeWithContentValues’  in the reactions.js to compatible customized built-in element

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/26327)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/356725793)

File: custom-elements/reactions/HTMLButtonElement.html [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+testReflectAttrWithDepAttr('formEnctype', 'formenctype', 'type', 'text/plain', 'multipart/form-data', 'submit', 'formEnctype on HTMLButtonElement', 'button', 'form', HTMLButtonElement);
+testReflectAttrWithDepAttr('formMethod', 'formmethod', 'type', 'get', 'post', 'submit', 'formMethod on HTMLButtonElement', 'button', 'form', HTMLButtonElement);
+testReflectAttrWithContentValuesAndDepAttr('formNoValidate', 'formnoValidate', 'type', true, 'true', false, 'false', 'submit', 'formNoValidate on HTMLButtonElement', 'button', 'form', HTMLButtonElement);
+testReflectAttrWithDepAttr('formTarget', 'formtarget', 'type', '_blank', '_self', 'submit', 'formTarget on HTMLButtonElement', 'button', 'form', HTMLButtonElement);
+
+//In parent node, sub node's observeAttribute can enqueue by changing attribute value 
+//Test reflect attribute with content values and parent node
+function testReflectAttrWithContentValuesAndPNode(jsAtName, coAtName, jsAtValue1, coAtValue1, jsAtValue2, coAtValue2, name, elementName, pElementName, interfaceName) {
```
[the remark this answers is not in the document]
Comment (Honry) [reply custom-elements/reactions/HTMLButtonElement.html]: @rniwa, @xiaoyu3x is no long active in w-p-t, I just push a new commit that addresses your comment. PTAL.

File: custom-elements/reactions/resources/reactions.js [hunk 2]
```diff
@@ -126,32 +126,42 @@ function testCloner(testFunction, name) {
     }, name + ' must enqueue an attributeChanged reaction when cloning an element only for observed attributes');
 }
 
-function testReflectAttributeWithContentValues(jsAttributeName, contentAttributeName, validValue1, contentValue1, validValue2, contentValue2, name) {
+function testReflectAttributeWithContentValues(jsAttributeName, contentAttributeName, validValue1, contentValue1, validValue2, contentValue2, name, elementName, interfaceName) {
     test(function () {
-        var element = define_new_custom_element([contentAttributeName]);
-        var instance = document.createElement(element.name);
+        if(elementName === undefined) {
```
Comment (rniwa) [inline custom-elements/reactions/resources/reactions.js]: Nit: should have a space between `if` and `(`.

File: custom-elements/reactions/resources/reactions.js [hunk 3]
```diff
@@ -126,32 +126,42 @@ function testCloner(testFunction, name) {
...
+            var element = define_build_in_custom_element([contentAttributeName], interfaceName, elementName);
+            var instance = document.createElement(elementName, { is: element.name });
+        }
         assert_array_equals(element.takeLog().types(), ['constructed']);
         instance[jsAttributeName] = validValue1;
         var logEntries = element.takeLog();
         assert_array_equals(logEntries.types(), ['attributeChanged']);
 
-        assert_attribute_log_entry(logEntries.last(), {name: contentAttributeName, oldValue: null, newValue: contentValue1, namespace: null});
+        assert_attribute_log_entry(logEntries.last(), { name: contentAttributeName, oldValue: null, newValue: contentValue1, namespace: null });
```
Comment (rniwa) [inline custom-elements/reactions/resources/reactions.js]: Let's not needlessly modify the coding style of the test harness code.

File: custom-elements/reactions/resources/reactions.js [hunk 4]
```diff
@@ -126,32 +126,42 @@ function testCloner(testFunction, name) {
...
     }, name + ' must enqueue an attributeChanged reaction when adding ' + contentAttributeName + ' content attribute');
 
     test(function () {
-        var element = define_new_custom_element([contentAttributeName]);
-        var instance = document.createElement(element.name);
+        if(elementName === undefined) {
```
Comment (rniwa) [inline custom-elements/reactions/resources/reactions.js]: Ditto.

File: custom-elements/reactions/resources/reactions.js [hunk 5]
```diff
@@ -126,32 +126,42 @@ function testCloner(testFunction, name) {
...
+            var element = define_build_in_custom_element([contentAttributeName], interfaceName, elementName);
+            var instance = document.createElement(elementName, { is: element.name });
+        }
         instance[jsAttributeName] = validValue1;
         assert_array_equals(element.takeLog().types(), ['constructed', 'attributeChanged']);
         instance[jsAttributeName] = validValue2;
         var logEntries = element.takeLog();
         assert_array_equals(logEntries.types(), ['attributeChanged']);
-        assert_attribute_log_entry(logEntries.last(), {name: contentAttributeName, oldValue: contentValue1, newValue: contentValue2, namespace: null});
+        assert_attribute_log_entry(logEntries.last(), { name: contentAttributeName, oldValue: contentValue1, newValue: contentValue2, namespace: null });
```
Comment (rniwa) [inline custom-elements/reactions/resources/reactions.js]: Ditto.

File: custom-elements/reactions/resources/reactions.js (hunk 2 above)
Comment (Honry) [reply custom-elements/reactions/resources/reactions.js]: @rniwa, thanks for catching this, done.

File: custom-elements/reactions/resources/reactions.js (hunk 3 above)
Comment (Honry) [reply custom-elements/reactions/resources/reactions.js]: Fixed.

**Review by rniwa — APPROVED** [review]
Great. Thanks for the test!

**Pull request merged**